Sources/CasePaths/CasePathable.swift

```diff
@@ -277,28 +320,44 @@ extension CaseKeyPath {
   where Root == Case<Enum>, Value == Case<AssociatedValue> {
     rhs[case: lhs] != nil
   }
+
+  public func extract<R, V>(from root: R) -> V? where Root == Case<R>, Value == Case<V> {
+    Case(self).extract(from: root)
+  }
```

Comment:
Without a `CasePathable` requirement to tie a `[case:]` subscript to, we need a way to utilize this functionality more generally.